ipn/ipnlocal/cert.go

```diff
@@ -157,6 +170,42 @@ func shouldStartDomainRenewal(domain string, now time.Time, pair *TLSCertKeyPair
 	return false, nil
 }
 
+func (b *LocalBackend) shouldStartDomainRenewalByARI(cs certStore, now time.Time, pair *TLSCertKeyPair) (bool, error) {
+	var blocks []*pem.Block
+	rest := pair.CertPEM
+	for len(rest) > 0 {
+		var block *pem.Block
+		block, rest = pem.Decode(rest)
+		if block == nil {
+			return false, fmt.Errorf("parsing certificate PEM")
+		}
+		blocks = append(blocks, block)
+	}
+	if len(blocks) < 2 {
+		return false, fmt.Errorf("could not parse certificate chain from certStore, got %d PEM block(s)", len(blocks))
+	}
+	ac, err := acmeClient(cs)
+	if err != nil {
+		return false, err
+	}
+	ctx, cancel := context.WithTimeout(b.ctx, 5*time.Second)
+	defer cancel()
+	ri, err := ac.FetchRenewalInfo(ctx, blocks[0].Bytes, blocks[len(blocks)-1].Bytes)
```

Comment:
The last element of the slice is most likely to be the issuer when the length of the slice is 2, but if we found ourselves for some reason looking at a slice of 3+, I'd still think/guess that the 1th element is most likely to be the issuer of the 0th. (There's no particular guarantee of this — the only real check is to actually build the chain from the assumption that the slice is unordered.)

If we see a significant number of failures from `FetchRenewalInfo`, picking the wrong cert as the issuer might turn out to be the/a reason. In that case, we might to really build the chain.

For now, as a first version, I'd pass `blocks[1].Bytes` as the issuer and keep an eye on the failure log.

Reply:
Good point. I was specifically trying to get the root CA here, but an intermediate is the actual "issuer" in this case.
And yeah, the proper way would be to  parse everything and figure out which one is leaf and which one is issuer.

Updated to use `blocks[1]`